From 17df8b13132255c74cbdc369f39c56e3004301a0 Mon Sep 17 00:00:00 2001
From: Julian Scheel <julian@jusst.de>
Date: Mon, 23 Apr 2012 18:02:59 +0200
Subject: [PATCH] mpegtspacketizer: Disable reset of base_pcrtime.

As tsdemux.c creates continous pts in rollover cases the packetizer does not
need to care about this and thus does not need to reset base_pcrtime.
---
 gst/mpegtsdemux/mpegtspacketizer.c |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/gst/mpegtsdemux/mpegtspacketizer.c b/gst/mpegtsdemux/mpegtspacketizer.c
index 41ff6dd..c06160d 100644
--- a/gst/mpegtsdemux/mpegtspacketizer.c
+++ b/gst/mpegtsdemux/mpegtspacketizer.c
@@ -2853,6 +2853,7 @@ mpegts_packetizer_reset_skew (MpegTSPacketizer2 * packetizer)
   GST_DEBUG ("reset skew correction");
 }
 
+#if 0
 static void
 mpegts_packetizer_resync (MpegTSPacketizer2 * packetizer, GstClockTime time,
     GstClockTime gstpcrtime, gboolean reset_skew)
@@ -2870,7 +2871,7 @@ mpegts_packetizer_resync (MpegTSPacketizer2 * packetizer, GstClockTime time,
     packetizer->skew = 0;
   }
 }
-
+#endif
 
 /* Code mostly copied from -good/gst/rtpmanager/rtpjitterbuffer.c */
 
@@ -2968,6 +2969,7 @@ calculate_skew (MpegTSPacketizer2 * packetizer, guint64 pcrtime,
 
   if (G_LIKELY (gstpcrtime >= packetizer->base_pcrtime))
     send_diff = gstpcrtime - packetizer->base_pcrtime;
+#if 0
   else if (GST_CLOCK_TIME_IS_VALID (time)) {
     /* elapsed time at sender, timestamps can go backwards and thus be smaller
      * than our base time, take a new base time in that case. */
@@ -2980,6 +2982,9 @@ calculate_skew (MpegTSPacketizer2 * packetizer, guint64 pcrtime,
     /* at least try to get a new timestamp.. */
     packetizer->base_time = -1;
   }
+#endif
+  else
+  send_diff = 0;
 
   GST_DEBUG ("gstpcr %" GST_TIME_FORMAT ", buftime %" GST_TIME_FORMAT ", base %"
       GST_TIME_FORMAT ", send_diff %" GST_TIME_FORMAT,
@@ -3018,7 +3023,9 @@ calculate_skew (MpegTSPacketizer2 * packetizer, guint64 pcrtime,
   if (ABS (delta - packetizer->skew) > GST_SECOND) {
     GST_WARNING ("delta - skew: %" GST_TIME_FORMAT " too big, reset skew",
         GST_TIME_ARGS (delta - packetizer->skew));
+#if 0
     mpegts_packetizer_resync (packetizer, time, gstpcrtime, TRUE);
+#endif
     send_diff = 0;
     delta = 0;
   }
-- 
1.7.10

